src/internals/index.js

```diff
@@ -152,36 +155,27 @@ export function createStitches(config = {}) {
                 styleSheetKey = `${prop}_${propValue['@initial']}`;
               }
 
-              Object.entries(config.media).forEach(([key, val]) => {
-                const breakpoint = `@${key}`;
-
+              if (breakpoint) {
                 if (propValue[breakpoint] === undefined) return;
-
+                const val = config.media[mediaKey];
                 if (val === true) {
                   styleSheetKey = `${prop}_${propValue[breakpoint]}`;
                 } else if (typeof val === 'string') {
-                  const correctedWindowWidth =
-                    PixelRatio.getPixelSizeForLayoutSize(windowWidth); // eslint-disable-line
-
-                  // TODO: how do we quarantee the order of breakpoint matches?
-                  // The order of the media key value pairs should be constant
-                  // but is that guaranteed? So if the keys are ordered from
-                  // smallest screen size to largest everything should work ok...
-                  const match = utils.resolveMediaRangeQuery(
-                    val,
-                    correctedWindowWidth
-                  );
-
-                  if (match) {
-                    styleSheetKey = `${prop}_${propValue[breakpoint]}`;
-                  }
+                  styleSheetKey = `${prop}_${propValue[breakpoint]}`;
```

Comment:
This can also be simplified so that there is only one if:

```js
if (val === true || typeof val === 'string') {
  styleSheetKey = `${prop}_${propValue[breakpoint]}`;
}
```